### PR TITLE
`echidna.sh`

### PR DESCRIPTION
**Pull request by naszam**
- Add `echidna.sh` to address solc conflicts with nix and lock echidna to solc 0.6.12

### CODE REVIEW

**Review by gbalabasquer — APPROVED** [review]
Haven't tested not reviewed it in detail but not very relevant to stop merging it.

File: echidna/echidna.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                VEST=DssVestTransferrableEchidnaTest
+                ;;
+    *)          echo "Incorrect vest contract: '$1' (mintable, suckable, transferrable)"; exit 1
+                ;;
+esac
+
+SOLC=~/.solc-select/artifacts/solc-0.6.12
+
+# Check if solc-select is installed
+[[ -x solc-select ]] || pip3 install solc-select
```
Comment (brianmcmichael) [inline echidna/echidna.sh:23]: We probably need an update to the readme about these scripts, with a mention that the script may install software on the developer's machine. 

Alternatively, and probably preferably, we might want the script to fail here if it's not installed and tell the user to install solc-select with `pip3 install solc-select` to run the script.

File: echidna/echidna.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+[[ "$1" ]] || { echo "Please specify the vest contract to fuzz (e.g. mintable, suckable, transferrable)"; exit 1; }
+
+case "$1" in
+            mintable)
+                VEST=DssVestMintableEchidnaTest
+                ;;
+            suckable)
+                VEST=DssVestSuckableEchidnaTest
+                ;;
+    transferrable)
```
[the remark this answers is not in the document]
Comment (naszam) [reply echidna/echidna.sh:13]: should be done in [9f814ae](https://github.com/makerdao/dss-vest/pull/46/commits/9f814aee8ff58f5886091cfa071dc1ce5510b73a)

File: echidna/echidna.sh (hunk 1 above)
Comment (naszam) [reply echidna/echidna.sh]: I've switched to `duppsolc` to use DappTools solc static versions together with `echidnup` to install echidna via nix, and updated the readme accordingly, adding a check in the script to make sure `duppsolc` is installed ([1b02a80](https://github.com/makerdao/dss-vest/pull/46/commits/1b02a80c67fb0e0f6f713375c042846c7e4c8cc3), [ca0bf1f](https://github.com/makerdao/dss-vest/pull/46/commits/ca0bf1f17bb98f503d524c48b4572dbd1b5685ac))

File: README.md [hunk 3]
```diff
@@ -104,19 +104,20 @@ Allows governance to schedule a point in the future to end the vest. Used for pl
...
+- Install Echidna v1.7.3
   ```
-  $ nix-env -i -f https://github.com/crytic/echidna/tarball/master
+  $ echidnup v1.7.3
   ```
 
 ### Run Echidna Tests
 
-- Install solc 0.6.12:
+- Install [duppsolc](https://github.com/naszam/duppsolc#installing)
```
Comment (gbalabasquer) [inline README.md:115]: I'd place this as an alternative but would still leave the direct way.

File: echidna/echidna.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+             ;;
+esac
+
+SOLC=~/.nix-profile/bin/solc-0.6.12
+
+# Check if duppsolc is installed
+[[ -x "$(command -v duppsolc)" ]] || { echo "Please install duppsolc"; exit 1; }
+
+# Check if solc-0.6.12 is installed
+[[ -f "$SOLC" ]] || duppsolc 0.6.12
```
Comment (gbalabasquer) [inline echidna/echidna.sh:26]: As I said before, I'd still leave the way to do it directly and not force to install an extra dependency.

File: README.md (hunk 3 above)
Comment (naszam) [reply README.md]: Done in 1c432c8

File: echidna/echidna.sh (hunk 4 above)
Comment (naszam) [reply echidna/echidna.sh]: Done in 1c432c8

File: echidna/echidna.sh (hunk 1 above)
Comment (naszam) [reply echidna/echidna.sh]: Following @gbalabasquer review, I've kept the direct way of installing the dependencies and updated the script accordingly, leaving `echidnup` and `duppsolc` as optional in 1c432c8

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Review by brianmcmichael — APPROVED** [review]

**Pull request merged**